core/sodasql/cli/cli.py

```diff
@@ -488,3 +498,143 @@ def scan(scan_yml_file: str, warehouse_yml_file: str, variables: tuple, time: st
         logger.info(f'Exiting with code 1')
         sys.exit(1)
+
+
+@main.command(short_help="Ingest test information from different tools")
+@click.argument(
+    "tool",
+    required=True,
+    type=click.Choice(["dbt"]),
+)
+@click.option(
+    "--warehouse-yml-file",
+    help="The warehouse yml file.",
+    required=True,
+)
+@click.option(
+    "--dbt-manifest",
+    help="The path to the dbt manifest file",
+    default=None,
+    type=Path,
+)
+@click.option(
+    "--dbt-run-results",
+    help="The path to the dbt run results file",
+    default=None,
+    type=Path,
+)
+def ingest(
+    tool: str,
+    warehouse_yml_file: str,
+    dbt_manifest: Optional[Path],
+    dbt_run_results: Optional[Path],
+):
+    """
+    Ingest test information from different tools.
+
+    Arguments
+    ---------
+    tool : str
+        The tool name.
+    warehouse_yml_file : str
+        The warehouse yml file.
+    dbt_manifest : Optional[Path]
+        The path to the dbt manifest.
+    dbt_run_results : Optional[Path]
+        The path to the dbt run results.
+    """
+    logger.info(SODA_SQL_VERSION)
+    if tool == "dbt":
+        try:
+            from dbt.contracts.results import TestStatus
+            from dbt.contracts.graph.compiled import CompiledSchemaTestNode
+            from sodasql import dbt as soda_dbt
+        except ImportError as e:
+            raise RuntimeError(
+                "Soda SQL dbt extension is not installed: $ pip install soda-sql-dbt"
+            ) from e
+        if dbt_manifest is None:
+            raise ValueError(f"Dbt manifest is required: {dbt_manifest}")
+        if dbt_run_results is None:
+            raise ValueError(f"Dbt run results is required: {dbt_run_results}")
+
+        warehouse_yml_parser = build_warehouse_yml_parser(warehouse_yml_file)
+        warehouse_yml = warehouse_yml_parser.warehouse_yml
+
+        soda_server_client = create_soda_server_client(warehouse_yml)
+        if not soda_server_client.api_key_id or not soda_server_client.api_key_secret:
+            raise ValueError("Missing Soda cloud api key id and/or secret.")
+
+        with dbt_manifest.open("r") as manifest:
+            model_nodes, test_nodes = soda_dbt.parse_manifest(json.load(manifest))
+        with dbt_run_results.open("r") as run_results:
+            parsed_run_results = soda_dbt.parse_run_results(json.load(run_results))
+
+        models_with_tests = soda_dbt.create_models_to_tests_mapping(
+            model_nodes, test_nodes, parsed_run_results
+        )
+
+        tests_in_run_results = [
+            (run_result, test_nodes[run_result.unique_id])
+            for run_result in parsed_run_results
+            if run_result.unique_id in test_nodes.keys()
+        ]
+
+        tests = [
+            Test(
+                id=Parser.create_test_id(
+                    test_expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
```

Comment:
I do not understand how/why I get non-compiled nodes. I would expect that for tests being executed - and thus part of the run results - that they need to be compiled... I will have a further look at this.

Reply:
I do not have a complete answer for this yet, the `if` statement solves the issue though, so I am approaching this as non-urgent. Moreover, we need to decide what is a proper way to create the test id, as mentioned in one of the comments below, maybe we should remove the `compiled_sql` all together.